Log a message when initializing the WebContent process

pvollan opened the pull request:
#### c2a9f39f42b833b6c1640f954cfc06a712dd04cd
<pre>
Log a message when initializing the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=241829">https://bugs.webkit.org/show_bug.cgi?id=241829</a>

Reviewed by Geoffrey Garen.

Log a long message to make sure the XPC connection to the log daemon for oversized messages is opened.
This is needed to block launchd after the WebContent process has launched, since access to launchd is
required when opening new XPC connections.

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::initializeXPCConnectionToLogd):
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: <a href="https://commits.webkit.org/251892@main">https://commits.webkit.org/251892@main</a>
</pre>

geoffreygaren commented on Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
```diff
@@ -241,6 +241,14 @@ static void softlinkDataDetectorsFrameworks()
+    memset(stringWithSpaces, ' ', sizeof(stringWithSpaces));
+    stringWithSpaces[sizeof(stringWithSpaces) - 1] = 0;
+    WEBPROCESS_RELEASE_LOG(Process, "%p WebProcess::platformInitializeWebProcess %s", this, stringWithSpaces);
+
```
I'd suggest a helper function here because a helper function encapsulates the mechanism for initializing the log daemon. Encapsulation is always valuable for reducing complexity, and it is especially nice in cases like this, where the implementation is a bit surprising, and subject to change once we have explicit SPI for this.

Something like "void initializeXPCConnectionToLogd()".

pvollan replied:
> I'd suggest a helper function here because a helper function encapsulates the mechanism for initializing the log daemon. Encapsulation is always valuable for reducing complexity, and it is especially nice in cases like this, where the implementation is a bit surprising, and subject to change once we have explicit SPI for this.
> 
> Something like "void initializeXPCConnectionToLogd()".

Ah, yes, that is a good point! Fixed in latest patch.

Thanks for reviewing!

geoffreygaren reviewed (approved):
r=me

pvollan committed:
Log a message when initializing the WebContent process
https://bugs.webkit.org/show_bug.cgi?id=241829

Reviewed by Geoffrey Garen.

Log a long message to make sure the XPC connection to the log daemon for oversized messages is opened.
This is needed to block launchd after the WebContent process has launched, since access to launchd is
required when opening new XPC connections.

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::initializeXPCConnectionToLogd):
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: https://commits.webkit.org/251892@main
